src/routes/functions/components/FunctionDrawer.tsx

```diff
@@ -92,14 +84,11 @@ const FunctionDrawer = (): JSX.Element => {
         }
     };
 
-    const { updateNameDialog, updateNameButton } = useUpdateName({
+    const { updateNameDialog, updateNameButton } = useUpdateAssetName({
         dialogTitle: 'Rename function',
-        assetKey: func?.key ?? '',
-        assetName: func?.name ?? '',
-        assetUpdating: useAppSelector(
-            (state) => state.functions.functionUpdating
-        ),
-        updateSlice: updateFunction,
+        asset: func ?? null,
```

Comment:
You don't need the `?? null` here because func is already of type `FunctionT | null`